optax/_src/transform.py

```diff
@@ -748,6 +748,44 @@ def update_fn(updates, state, params=None):
   return base.GradientTransformation(init_fn, update_fn)
 
 
+class ScaleByRpropState(NamedTuple):
+  nu: base.Updates
+  prev_updates: base.Updates
+
+
+def scale_by_rprop(
+    learning_rate: float,
+    eta_minus: float = 0.5,
+    eta_plus: float = 1.2,
+    min_step_size: float = 1e-6,
+    max_step_size: float = 50.0,
+) -> base.GradientTransformation:
+
+  def init_fn(params):
+    nu = jax.tree_util.tree_map(
+      lambda p: learning_rate * jnp.ones_like(p), params)
+    prev_updates = jax.tree_util.tree_map(jnp.zeros_like, params)
+    return ScaleByRpropState(nu, prev_updates)
+
+  def update_fn(updates, state, params=None):
+    del params
+    sign = jax.tree_util.tree_map(
+      lambda g, prev_g: g * prev_g, updates, state.prev_updates)
+    nu = jax.tree_util.tree_map(
+      lambda s, nu: jnp.where(
+        s == 0,
+        nu,
+        jnp.clip(nu * jnp.where(s < 0, eta_plus, eta_minus),
+                 a_min=min_step_size, a_max=max_step_size)),
+      sign, state.nu)
+    updates = jax.tree_util.tree_map(
```

Comment:
In the ref, they reverse the previous update if the sign flips no? And they also reset the previous updates to be zero in that case.

Reply:
They do, I should have mentioned that I'm actually using the torch implementation as a reference; I thought it would be a more concrete baseline. Should I mention this in the docstring? Or is it better to stick to the paper?

Reply:
Ah! Yep if you could simply add it in the doctring that would be good.
By curiosity, has there been recent investigations of the algorithm and such variant?

Reply:
Thanks! I added it. There have been some [further investigations](https://www.sciencedirect.com/science/article/abs/pii/S0925231201007007) but I'm not aware of anything very recent.

Reply:
Yes, just for some context. You may add that paper even if it's a bit outdated, that's still interesting. After that, if you can squash all commits I think we'll be good.

Reply:
I just added the paper and squashed. Thank you for the review!